### ISSUE TITLE
No executable found matching command "dotnet-repl"

### ISSUE THREAD

**Issue by gavdraper** [dotnet/cli#1289]
After installing the latest .Net CLI for OS X the REPL has stopped working. I'm on OS X 10.11 and get the following output

```
Gavins-MacBook-Air:~ gavindraper$ dotnet repl
No executable found matching command "dotnet-repl"
```

This worked fine before installing the latest pkg

**Comment by BusyJay** [comment]
Now it complains

    % dotnet repl
    No executable found matching command "dotnet-repl-csi"

Though I can launch repl by `csi`。

**Comment by gavinbeatty** [comment]
Same issue.

````
Microsoft Windows [Version 6.1.7601]
Copyright (c) 2009 Microsoft Corporation.  All rights reserved.

C:\Users\gavinbeatty>dotnet repl
No executable found matching command "dotnet-repl-csi"

C:\Users\gavinbeatty>csi
Microsoft (R) Visual C# Interactive Compiler version 1.3.0.60225
Copyright (C) Microsoft Corporation. All rights reserved.

Type "#help" for more information.
> var s = "hello";
> s
"hello"
> ^C
C:\Users\gavinbeatty>dotnet help
.NET Command Line Tools (1.0.0-beta-001598)
Usage: dotnet [common-options] [command] [arguments]

Arguments:
  [command]     The command to execute
  [arguments]   Arguments to pass to the command

Common Options (passed before the command):
  -v|--verbose  Enable verbose output
  --version     Display .NET CLI Version Info

Common Commands:
  new           Initialize a basic .NET project
  restore       Restore dependencies specified in the .NET project
  build         Builds a .NET project
  publish       Publishes a .NET project for deployment (including the runtime)
  run           Compiles and immediately executes a .NET project
  repl          Launch an interactive session (read, eval, print, loop)
  pack          Creates a NuGet package

C:\Users\gavinbeatty>dotnet --version
.NET Command Line Tools (1.0.0-beta-001598)

Product Information:
 Version:     1.0.0-beta-001598
 Commit Sha:  7582649f88

Runtime Environment:
 OS Name:     Windows
 OS Version:  6.1.7601
 OS Platform: Windows
 Runtime Id:  win7-x64

C:\Users\gavinbeatty>
````

**Comment by vquaiato** [comment]
Is this a bug?

I am having the same issue, on OSX:

```
System Software Overview:
      System Version: OS X 10.10.5 (14F1021)
      Kernel Version: Darwin 14.5.0
      Secure Virtual Memory: Enabled
      Time since boot: 2 days 15:29
```

```
➜  dotnet
.NET Command Line Tools (1.0.0-beta-001598)
```
````
➜ dotnet repl                       
No executable found matching command "dotnet-repl-csi"
```

Any clues?

**Comment by IgorKochetov** [comment]
Same issue:
`Igors-MacBook-Pro:hwapp IgorK$ dotnet --version
.NET Command Line Tools (1.0.0-beta-001571)

Product Information:
 Version:     1.0.0-beta-001571
 Commit Sha:  0818068e0b

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  10.10
 OS Platform: Darwin
 Runtime Id:  osx.10.10-x64
Igors-MacBook-Pro:hwapp IgorK$ dotnet repl
No executable found matching command "dotnet-repl-csi"
Igors-MacBook-Pro:hwapp IgorK$ csi
Microsoft (R) Visual C# Interactive Compiler version 1.2.0.60202
Copyright (C) Microsoft Corporation. All rights reserved.

Type "#help" for more information.
`

**Comment by adamralph** [comment]
Same issue, with a fresh installation.

**Comment by myoung34** [comment]
Same on ubuntu 15.10

```
$ dotnet --version
.NET Command Line Tools (1.0.0-beta-001675)

Product Information:
 Version:     1.0.0-beta-001675
 Commit Sha:  N/A

Runtime Environment:
 OS Name:     ubuntu
 OS Version:  15.10
 OS Platform: Linux
 Runtime Id:  ubuntu.15.10-x64
$ dotnet repl
No executable found matching command "dotnet-repl-csi"

```

**Comment by jamesqo** [comment]
I also seem to be getting this issue with `dotnet compile`, with the freshest build provided by the link on the README.

Output of `dotnet --version`:

```
.NET Command Line Tools (1.0.0-beta-001840)

Product Information:
 Version:     1.0.0-beta-001840
 Commit Sha:  N/A

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.10586
 OS Platform: Windows
 Runtime Id:  win10-x64
```

**Comment by jamesqo** [comment]
Just tested from an Ubuntu 14.04 VM on [Cloud9](http://c9.io) using the [Getting Started](http://dotnet.github.io/getting-started/) instructions from the website, seems to be happening there too.

**Comment by blackdwarf** [comment]
All, REPL is not going to be distributed with the CLI core package. 

@jamesqo are you still seeing the behavior? I cannot repro with the latest bits?

**Comment by adamralph** [comment]
@blackdwarf are you going to remove the REPL line from the help output? I guess that is what is generating the expectation right now.

**Comment by jamesqo** [comment]
@blackdwarf Yes, it looks like it's still happening with the latest version, with both `compile` and `repl-csi`. (You can try it yourself at the link to Cloud9 I posted.)

**Comment by naamunds** [comment]
> REPL is not going to be distributed with the CLI core package.

@blackdwarf Will it be distributed in a different package for rc2, or has it been cut altogether?  Also, it seems like it should be removed from the "help" output either way.

**Comment by MichaelSimons** [comment]
@naamunds There is already a PR (https://github.com/dotnet/cli/pull/1530) to remove REPL from the help.

**Comment by adamralph** [comment]
:confused:

**Comment by blackdwarf** [comment]
Thanks @MichaelSimons. @naamunds REPL was not "cut" in a sense that it was discontinued. We are working towards having a story on how to get it in the CLI via the existing extensibility mechanisms (e.g. "tools") but  at this point we are heads down doing other stuff, so the work is standing still.

**Comment by blitzd** [comment]
First exposure to dotnet core I run the command line and think 'oh how nice, a REPL'. Then the error, then googling the error, then reading through this and the PR and... ah yes, there's the MS that I know!

If you can't be bothered to update the usage to reflect the actual functionality that is present (for months on end, even), how do you think that looks to people who are just picking up these tools for the first time?

**Comment by blackdwarf** [comment]
Moving this to RC2 to track removal of "repl" from the command help. This removal is tracked in the #1530 PR.

**Comment by piotrpMSFT** [comment]
Fixed by #1530
